File: lib/features/admin/main_data/units/data/models/polsek_model.dart
import 'polres_model.dart';
import 'wilayah_model.dart';

class PolsekModel {
  final int id;
  final String namaPolsek;
  final String kapolsek;
  final String noTelp;
  final String kode;
  final int polresId;
  final int wilayahId;
  final PolresModel? polres;   // Optional (Preload)
  final WilayahModel? wilayah; // Optional (Preload)

  PolsekModel({
    required this.id,
    required this.namaPolsek,
    required this.kapolsek,
    required this.noTelp,
    required this.kode,
    required this.polresId,
    required this.wilayahId,
    this.polres,
    this.wilayah,
  });

  factory PolsekModel.fromJson(Map<String, dynamic> json) {
    return PolsekModel(
      id: json['id_polsek'] ?? 0,
      namaPolsek: json['nama_polsek'] ?? '',
      kapolsek: json['kapolsek'] ?? '',
      noTelp: json['no_telp_polsek'] ?? '',
      kode: json['kode'] ?? '',
      polresId: json['id_polres'] ?? 0,
      wilayahId: json['id_wilayah'] ?? 0,
      polres: json['polres'] != null ? PolresModel.fromJson(json['polres']) : null,
      wilayah: json['wilayah'] != null ? WilayahModel.fromJson(json['wilayah']) : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id_polsek': id,
      'nama_polsek': namaPolsek,
      'kapolsek': kapolsek,
      'no_telp_polsek': noTelp,
      'kode': kode,
      'id_polres': polresId,
      'id_wilayah': wilayahId,
    };
  }
}

File: go-backend-auth/models/polsekModel.go
package models

import "time"

type Polsek struct {
	ID           uint   `gorm:"primaryKey;column:id_polsek" json:"id_polsek"`
	NamaPolsek   string `gorm:"type:varchar(50);not null" json:"nama_polsek"`
	Kapolsek     string `gorm:"type:varchar(50);not null" json:"kapolsek"`
	NoTelpPolsek string `gorm:"type:varchar(20)" json:"no_telp_polsek"`
	Kode         string `gorm:"type:varchar(20)" json:"kode"`

	// Foreign Key ke Polres (Hirarki Komando)
	PolresID  uint `gorm:"column:id_polres;type:int;not null" json:"id_polres"`
	Polres   Polres `gorm:"foreignKey:PolresID;references:ID" json:"polres,omitempty"`

	// Foreign Key ke Wilayah (Lokasi Fisik)
	WilayahID uint `gorm:"column:id_wilayah;type:int;not null" json:"id_wilayah"`
	Wilayah   Wilayah `gorm:"foreignKey:WilayahID;references:ID" json:"wilayah,omitempty"`

	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

func (Polsek) TableName() string {
	return "polsek"
}